relay-server/src/services/processor/span/processing.rs

```diff
@@ -346,7 +350,9 @@ pub fn extract_from_event(
 
     add_span(transaction_span.into());
 
-    state.spans_extracted = true;
+    if let Some(headers) = state.event_headers.as_mut() {
```

Comment:
What if it is not `None`, we should still set it to true?

Reply:
I realized I can keep the reference from the beginning of the function alive and make it mutable.

Semantically, `state.event_headers` cannot be `None` if `state.event` is not empty, so my first attempt at this PR was to put them in an `Option` together. But that made event handling more complicated because of the two option levels (`Option(Annotated(Option(Event), _))`). Long term, I believe we should remove both from the state and pass them as plain old variables, as you suggested.